pkg/controllers/dynakube/extension/databases/deployment.go

```diff
@@ -0,0 +1,270 @@
+			PeriodSeconds:       5,
+			TimeoutSeconds:      2,
+			FailureThreshold:    3,
+			SuccessThreshold:    1,
+		},
+		Resources:       buildContainerResources(dbex.Resources),
+		SecurityContext: buildContainerSecurityContext(),
+		VolumeMounts:    buildVolumeMounts(dbex),
+	}
+
+	return container
+}
+
+func buildContainerArgs(dk *dynakube.DynaKube) []string {
+	return []string{
+		"--podid=$(POD_NAME)",
+		fmt.Sprintf("--url=https://%s:%d", dk.Extensions().GetServiceNameFQDN(), consts.ExtensionsDatasourceTargetPort),
+		"--idtoken=" + tokenMountPath + "/" + tokenVolumeName,
+	}
+}
+
+func buildContainerEnvs() []corev1.EnvVar {
+	return []corev1.EnvVar{
+		{
+			Name: "POD_NAME",
+			ValueFrom: &corev1.EnvVarSource{
+				FieldRef: &corev1.ObjectFieldSelector{
+					FieldPath: "metadata.name",
+				},
+			},
+		},
+	}
+}
+
+func buildVolumeMounts(dbex extensions.DatabaseSpec) []corev1.VolumeMount {
+	volumeMounts := []corev1.VolumeMount{
+		{
+			Name:      userDataVolumeName,
+			MountPath: userDataMountPath,
+		},
+		{
+			Name:      tokenVolumeName,
+			MountPath: tokenMountPath,
+			ReadOnly:  true,
+		},
+		{
+			Name:      certsVolumeName,
+			MountPath: certsMountPath,
+			ReadOnly:  true,
+		},
+	}
+
+	return append(volumeMounts, dbex.VolumeMounts...)
+}
+
+func buildVolumes(dk *dynakube.DynaKube, dbex extensions.DatabaseSpec) []corev1.Volume {
+	volumes := []corev1.Volume{
+		{
+			Name: userDataVolumeName,
+			VolumeSource: corev1.VolumeSource{
+				EmptyDir: &corev1.EmptyDirVolumeSource{},
+			},
+		},
+		{
+			Name: tokenVolumeName,
+			VolumeSource: corev1.VolumeSource{
+				Secret: &corev1.SecretVolumeSource{
+					SecretName: dk.Extensions().GetTokenSecretName(),
+					Items: []corev1.KeyToPath{
+						{
+							Key:  consts.DatasourceTokenSecretKey,
+							Path: tokenVolumeName,
+						},
+					},
+				},
+			},
+		},
+		{
+			Name: certsVolumeName,
+			VolumeSource: corev1.VolumeSource{
+				Secret: &corev1.SecretVolumeSource{
+					SecretName: dk.Extensions().GetTLSSecretName(),
+					Items: []corev1.KeyToPath{
+						{
+							Key:  consts.TLSCrtDataName,
+							Path: consts.TLSCrtDataName,
+						},
+					},
+				},
+			},
+		},
+	}
+
+	return append(volumes, dbex.Volumes...)
+}
+
+func buildContainerResources(custom *corev1.ResourceRequirements) corev1.ResourceRequirements {
+	if custom != nil {
+		return *custom
+	}
+
+	return corev1.ResourceRequirements{
+		Requests: corev1.ResourceList{
+			corev1.ResourceMemory: resource.MustParse("256Mi"),
+			corev1.ResourceCPU:    resource.MustParse("250m"),
+		},
+		Limits: corev1.ResourceList{
+			corev1.ResourceMemory: resource.MustParse("512Mi"),
+			corev1.ResourceCPU:    resource.MustParse("500m"),
+		},
+	}
+}
+
+func buildPodSecurityContext() *corev1.PodSecurityContext {
+	return &corev1.PodSecurityContext{
+		SeccompProfile: &corev1.SeccompProfile{
+			Type: corev1.SeccompProfileTypeRuntimeDefault,
+		},
+		RunAsNonRoot: ptr.To(true),
+		RunAsGroup:   ptr.To(int64(1000)),
```

Comment:
i would prefer to extract it to consts. similar to https://github.com/dynatrace/dynatrace-operator/blob/ad61d9c9bfa827df7e19fcc92cdfcd5aa320fc95/pkg/controllers/dynakube/extension/eec/statefulset.go#L75